figma-shop/src/app/shared/services/order.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@angular/core';
+import { ProductInterface } from '../interfaces/product.interface';
+import { Subject } from 'rxjs';
+import { Router } from '@angular/router';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class OrderService {
+  ordersInCart: Subject<number> = new Subject<number>();
```

Comment:
public/private ?
Because of this `Subject` you don't see the first value when you subscribe for this, change it to `BehaviorSubject`, and you will be happy :)

Reply:
thanks =)